Sources/TuistGenerator/Generator/DescriptorGenerator.swift

```diff
@@ -67,7 +67,8 @@ public final class DescriptorGenerator: DescriptorGenerating {
     private let workspaceGenerator: WorkspaceGenerating
     private let projectGenerator: ProjectGenerating
 
-    public convenience init(defaultSettingsProvider: DefaultSettingsProviding = DefaultSettingsProvider()) {
+    public convenience init(defaultSettingsProvider: DefaultSettingsProviding = DefaultSettingsProvider(),
```

Comment:
Seeing `versionsFetcher` is passed in the init here, there isn't a need to pass it as a parameter to all the method calls - we could even pass `Versions` in the init as its fairly constant (not something that changes mid generation).

Reply:
Good catch. The idea was to only use the `VersionFetcher` from `TuistKit` and pass the version from there. Since we might need to use the version from other places, not just the generation logic, I don't think it needs to be implicitly obtained from the `DescriptorGenerator`.